ubuntudesign/docker-devrun
# Replace example Makefile with a "run" script

I decided that the `make run` interface is rather clunky (options had to be passed in as environment variables), and now basically redundant, so I'm removing the `Makefile.example`.

I've replaced it with a `./run.example` script, which takes the following optional arguments (which used to be passed with environment variables):

```
--port  - The port to run the service on
--debug  - Enable/disable Django DEBUG
--secret-key  - Provide a Django SECRET_KEY setting
```

QA
---

Find one of our existing Django projects, and download the `run` script into it:

``` bash
wget -O - https://raw.githubusercontent.com/nottrobin/docker-devrun/replace-makefile-with-script/run.example > run
chmod +x run
```

If it's a database-backed project, you'll also need to do `echo "DB=true" >> .env`.

Then try running said Django project with the new script:

``` bash
./run
```

And check it successfully runs the project on port `8000`.

Now try:

``` bash
./run commands  # List commands
./run --port 7654  # Run on a different port
```

*NB:* This doesn't currently build Sass.
